sea-orm-macros/src/derives/active_enum_display.rs

```diff
@@ -0,0 +1,129 @@
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote, quote_spanned};
+use syn::{LitStr, Lit};
+
+enum Error {
+    InputNotEnum,
+    Syn(syn::Error),
+    // TT(TokenStream),
+}
+
+struct Display {
+    ident: syn::Ident,
+    name: String,
+    variants: Vec<DisplayVariant>,
+}
+
+struct DisplayVariant {
+    ident: syn::Ident,
+    display_value: Option<LitStr>,
+}
+
+impl Display {
+    fn new(input: syn::DeriveInput) -> Result<Self, Error> {
+        let ident = input.ident;
+
+        let mut name = ident.to_string();
+
+        
+        let variant_vec = match input.data {
+            syn::Data::Enum(syn::DataEnum { variants, .. }) => variants,
+            _ => return Err(Error::InputNotEnum),
+        };
+
+        
+        let mut variants = Vec::new();
+        for variant in variant_vec {
+            let mut display_value = None;
+            let variant_span = variant.ident.span();
+            for attr in variant.attrs.iter() {
+                if !attr.path().is_ident("sea_orm") {
+                    continue;
+                }
+                attr.parse_nested_meta(|meta| {
+                    if meta.path.is_ident("display_value") {
+                        display_value = Some(meta.value()?.parse::<LitStr>()?);
+                    } else {let _other_value = Some(meta.value()?.parse::<Lit>()?);}
+
+                    Ok(())
+                })
+                .map_err(Error::Syn)?;
+            }
+
+            
+            variants.push(DisplayVariant {
+                ident: variant.ident,
+                display_value,
+            });
+        }
+        Ok(Display {
+            ident,
+            name,
+            variants,
+        })
+    }
+
+    fn expand(&self) -> syn::Result<TokenStream> {
+        let expanded_impl_active_enum = self.impl_active_enum();
+
+        Ok(expanded_impl_active_enum)
+    }
+
+    fn impl_active_enum(&self) -> TokenStream {
+        let Self {
+            ident,
+            name,
+            variants
+        } = self;
+
+        let variant_idents: Vec<syn::Ident> = variants
+            .iter()
+            .map(|variant| variant.ident.clone())
+            .collect();
+
+        let variant_display: Vec<TokenStream> = variants
+            .iter()
+            .map(|variant| {
+                if let Some(display_value) = &variant.display_value {
+                    let string = display_value.value();
+                    quote! { #string }
+                } else {
+                    let ident = &variant.ident;
+                    quote! { #ident }
+                }
+            })
+            .collect();
+
+        quote!(
+            impl #ident {
```

Comment:
My guess on the source line for `cannot find value in scope` error
I guess the issue is that the ident was somehow moved and therefore cannot be used 
However, using Tokenstream to store the info directly seems to have solved it